tracee: 0.11.0 -> 0.13.0

06kellyjac opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Updated tracee to 0.13.0

Also split out into separate outputs and dropped tracee-*
binaries due to the new single tracee binary

![image](https://user-images.githubusercontent.com/9866621/230032356-8c91a09c-0b5e-4258-b27e-7ce3047404ba.png)


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [X] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/1937

jsoo1 commented on nixos/tests/tracee.nix:
```diff
@@ -7,11 +7,11 @@ import ./make-test-python.nix ({ pkgs, ... }: {
       virtualisation.docker.enable = true;
 
-      environment.systemPackages = [
+      environment.systemPackages = with pkgs; [
```
I'm not 100% sure but I think the goal is to reduce the use of `with`?

SuperSandro2000 replied:
*over big code parts and entire files

06kellyjac replied:
In this case it's pretty clear which values are coming from the `with pkgs;` because there's no other `with`s and it's over only this array/list

SuperSandro2000 commented on pkgs/tools/security/tracee/default.nix:
```diff
@@ -59,27 +56,33 @@ buildGoModule rec {
   '';
 
-  doInstallCheck = true;
+  doInstallCheck = false;
```
Isn't this the default?

06kellyjac replied:
yeah, this is left over from debugging where I knew they changed the names of some of the binaries & didn't want to fail the build before I could modify.
Need to revert

SuperSandro2000 commented on pkgs/tools/security/tracee/default.nix:
```diff
@@ -59,27 +56,33 @@ buildGoModule rec {
 
-    mv ./dist/rules $out/share/tracee/
-    mv ./cmd/tracee-rules/templates $out/share/tracee/
+    ln -s $bin/bin/tracee $out/bin
```
This is not really how we are rushing multiple outputs

06kellyjac replied:
rushing?

SuperSandro2000 commented on pkgs/tools/security/tracee/default.nix:
```diff
@@ -59,27 +56,33 @@ buildGoModule rec {
   # see passthru.tests.integration
   doCheck = false;
 
+  outputs = [ "out" "bin" "legacy" "lib" "share" ];
```
```suggestion
  outputs = [ "out" "lib" ];
```

That would probably easier to consume

06kellyjac replied:
I'm not sure how the split makes it harder to consume?

I kept the main "out" very useful having pretty much everything a user will need but giving separate outputs means if you know you only need the main `tracee` binary it's there under `tracee.bin`
Legacy output will disappear once that's removed in the future. Upstream maintainers have given the greenlight to remove it now though & shouldn't be a problem for our unstable channel so I can drop that early

I guess rather than linking I could depend on `outputsToInstall`?

jsoo1 replied:
The main benefit of multiple outputs is mostly to reduce closure size. If the separate outputs are links to the bigger one then they can't be installed separately. So it is more complicated without providing much gain. I think I'd prefer just `out` and `lib` too.

06kellyjac replied:
sure it can, you can just include `tracee.bin` in your system env or container or whatever

```
λ nix bundle --bundler github:NixOS/bundlers#toDockerImage github:06kellyjac/nixpkgs/tracee#tracee.bin

λ mv tracee-0.12.0.tar.gz tracee-bin-0.12.0.tar.gz

λ nix bundle --bundler github:NixOS/bundlers#toDockerImage github:06kellyjac/nixpkgs/tracee#tracee

λ du $(readlink ./tracee-0.12.0.tar.gz)
58616   /nix/store/1qsd41aq064wxfmlwpi4z5w6g9395b3r-tracee-0.12.0.tar.gz

λ du $(readlink ./tracee-bin-0.12.0.tar.gz)
32668   /nix/store/cc18zmmb9dkgg00mzyvlxlj1clrvagxb-tracee-0.12.0.tar.gz

λ docker load < tracee-0.12.0.tar.gz
2013c20b6873: Loading layer [==================================================>]  1.812MB/1.812MB
a9cb236e24b5: Loading layer [==================================================>]    297kB/297kB
6cf112bf428d: Loading layer [==================================================>]  30.77MB/30.77MB
90fce17ae0cf: Loading layer [==================================================>]  8.202MB/8.202MB
47e670efd9ae: Loading layer [==================================================>]  143.4kB/143.4kB
3ec45d13d845: Loading layer [==================================================>]  92.16kB/92.16kB
185b597ab364: Loading layer [==================================================>]  839.7kB/839.7kB
5cda2fe2b3ab: Loading layer [==================================================>]  1.126MB/1.126MB
fa4ea5799bfd: Loading layer [==================================================>]  3.666MB/3.666MB
51d514292f31: Loading layer [==================================================>]  1.772MB/1.772MB
58f6e5d6835f: Loading layer [==================================================>]  583.7kB/583.7kB
0276fd48123f: Loading layer [==================================================>]  122.9kB/122.9kB
d87b026ad8ad: Loading layer [==================================================>]  2.888MB/2.888MB
a119ce264191: Loading layer [==================================================>]  53.23MB/53.23MB
9fe0399b3a83: Loading layer [==================================================>]  77.61MB/77.61MB
2f15ef34242c: Loading layer [==================================================>]     10MB/10MB
a1d3f83dd160: Loading layer [==================================================>]  10.24kB/10.24kB
ddee65b264cb: Loading layer [==================================================>]  10.24kB/10.24kB
a4b9d21d882f: Loading layer [==================================================>]  10.24kB/10.24kB
Loaded image: tracee-0.12.0:latest

λ docker run -it tracee-0.12.0:latest tracee --version
Tracee version "v0.12.0"
λ docker run -it tracee-0.12.0:latest tracee-ebpf --version
Tracee version "v0.12.0"

λ docker load < ./tracee-bin-0.12.0.tar.gz
58f6e5d6835f: Loading layer [==================================================>]  583.7kB/583.7kB
51d514292f31: Loading layer [==================================================>]  1.772MB/1.772MB
0276fd48123f: Loading layer [==================================================>]  122.9kB/122.9kB
d87b026ad8ad: Loading layer [==================================================>]  2.888MB/2.888MB
a119ce264191: Loading layer [==================================================>]  53.23MB/53.23MB
faecf4444776: Loading layer [==================================================>]  10.24kB/10.24kB
The image tracee-0.12.0:latest already exists, renaming the old one with ID sha256:616a58ae3275261d869b6be79c8d6dc4cee891e2c9fe5ffeaf98bd8f102dde07 to empty string
Loaded image: tracee-0.12.0:latest

λ docker run -it tracee-0.12.0:latest tracee --version
Tracee version "v0.12.0"
λ docker run -it tracee-0.12.0:latest tracee-ebpf --version
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "tracee-ebpf": executable file not found in $PATH: unknown.
ERRO[0000] error waiting for container: context canceled
```

06kellyjac replied:
I have dropped the legacy bins because the tracee team advised it. Now there are less outputs but it's staying as-is. `tracee` can be used perfectly fine without the share dir, it's just a pack of rules that people may or may not use but it will come by default via `outputsToInstall` instead of linking now

06kellyjac commented:
arm64 is now supported officially in 0.13.0 so I'll give that a go

06kellyjac committed:
tracee: 0.11.0 -> 0.13.0

Also split out into separate outputs and dropped tracee-*
binaries due to the new single tracee binary

Mark aarch64-linux as officially supported
